core/variable.js

```diff
@@ -0,0 +1,69 @@
+/**
```

Comment:
We already have core/variables.js and blocks/variables.js.  This file should have a more descriptive name--possibly variable_model.js.  Same for the class.